magma/backend/mlir/hardware_module.py

```diff
@@ -613,6 +620,39 @@ def make_register(T, data, init, result):
             make_register(T_i, data_i, init_i, result_i)
         return True
 
+    @wrap_with_not_implemented_error
+    def visit_magma_xmr_sink(self, module: ModuleWrapper) -> bool:
+        inst = module.module
+        defn = type(inst)
+        assert isinstance(defn, XMRSink)
+        xmr = defn.value
+        try:
+            self._ctx.parent.xmr_paths[xmr]
+        except KeyError:
+            pass
+        else:
+            return True
+        paths = get_xmr_paths(self._ctx, xmr)
+        assert len(paths) == len(module.operands)
+        self._ctx.parent.xmr_paths[xmr] = paths
+        return True
```

Comment:
Could we just move the code inside KeyError?
```suggestion
        try:
            self._ctx.parent.xmr_paths[xmr]
        except KeyError:
            paths = get_xmr_paths(self._ctx, xmr)
            assert len(paths) == len(module.operands)
            self._ctx.parent.xmr_paths[xmr] = paths
        return True
```

Reply:
semantically, yes I think that would be fine. Just a style preference to do it outside, with an early return in the no-exception case.